draft-ietf-httpbis-http2-secondary-certs.md

```diff
@@ -753,6 +753,80 @@ any stream which references the failing certificate in question or process the
 requests as unauthenticated and provide error information at the HTTP semantic
 layer.
 
+# Certificate Extensions {#extension}
+
+## Allowed Primary Domain Certificate Extension {#extension-primary}
+
+The Allowed Primary Domain extension allows certificates to limit their use with
+Secondary Certificate Authentication.  The identities listed in this extension
+are restrictions asserted by the requester of the certificate and are not
+verified by the CA.  Conforming CAs SHOULD mark the allowedPrimaryDomain
+extension as non-critical.
+
+Each allowed primary domain is represented as an IA5String. The name MUST be in
+the "preferred name syntax", as specified by Section 3.5 of {{!RFC1034}} and as
+modified by Section 2.1 of {{!RFC1123}}.  Note that while uppercase and
+lowercase letters are allowed in domain names, no significance is attached to
+the case. In addition, while the string " " is a legal domain name,
+affiliatedDomain extensions MUST NOT contain " ".  Rules for encoding
+internationalized domain names are specified in Section 7.2 of {{!RFC5280}}.
+
+If the affiliatedDomain extension is present, the sequence MUST contain at least
+one entry.  Unlike the subject field, conforming CAs MUST NOT issue certificates
+with an affiliatedDomain extension containing empty GeneralName fields.  Clients
+that encounter such a certificate when processing a certification path MUST
+ignore that extension.
```

Comment:
I know that the effect of tolerance here is disabling the feature, but do we really need to mandate this particular form of tolerance?  Wouldn't it be better to consider the certificate invalid?  That is, if the extension is present and understood, it is an error for it to violate the rules.  That way we avoid weirdnesses accumulating.

Reply:
I'm okay with tightening here, but this seems at odds with importing the full set of SAN rules by reference.  I'd rather just state the rules for this extension than be "like SAN, except...."